force-app/main/default/classes/SOQL.cls

```diff
@@ -344,18 +413,26 @@ public with sharing class SOQL {
      * Example: setPrimaryContact() should set the 'Primary_Contact__c' field for every Opportunity in the org
      */
     public static void setPrimaryContact() {
-        //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
-        // Create a set of Account Ids from the Opportunities
-
-        // Query the Contacts
-
-        // Loop through each Contact and add the Account Id and Contact to the map
-        
-        // Loop through each Opportunity
-
-                // Update the Opportunity primary contact from accountToContactMap
-
-        // Update the Opportunities
+        List<Opportunity> oppList = getOpportunityDetailsForClosedWon();   //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
+        Set<Id> accountIdSet = new Set<Id>();
+        Map<Id, Contact> contactAccountMap = new Map<Id, Contact>();
+
+        for (Opportunity opp : oppList){        // Create a set of Account Ids from the Opportunities
+            if (opp.AccountId != null){
+                accountIdSet.add(opp.AccountId);
+            }
+        }        
+        List<Contact> contactList = [SELECT Id, Name, CreatedDate, AccountId        
+                                    FROM Contact 
+                                    WHERE AccountId 
+                                    IN :accountIdSet];         
+        for (Contact cont : contactList){// Loop through each Contact and add the Account Id and Contact to the map  
+            contactAccountMap.put(cont.AccountId, cont);
```

Comment:
What happens here if more than one Contact is associated with an Account? How could you order your `contactList` to associate the right Contact with this Account?